src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -362,30 +368,17 @@ def __getitem__(self, key: SupportsIndex | datetime | slice) -> float | ArrayLik
             A float if the key is a number or a timestamp.
             an numpy array if the key is a slice.
         """
+        if isinstance(key, slice):
+            assert key.step is None or key.step == 1
```

Comment:
Normally `__getitem__` must support different steps, so we should at least document this limitation in the method's docs. My intuition says we should probably raise a `ValueError` instead too, because the `__getitem__` interface allows for a different step, but at the end it is a programming error to pass a knowingly wrong value, so `assert` should be also OK.

PS: I just checked the [`__getitem__` docs](https://docs.python.org/3/reference/datamodel.html#object.__getitem__) and it doesn't impose anything when the value is unsupported, so I guess we are free to use `assert` or `ValueError`.

Reply:
Raising is probably better since it's a valid usage which is just not supported but can be reasonably expected by the user. Wouldn't an `IndexError` be more appropriate though?